internal/monitors/statsd/metadata.yaml

```diff
@@ -0,0 +1,75 @@
+- dimensions: null
+  doc: |
+
+
+    ### USAGE
+
+    #### Adding dimensions to StatsD metrics
+
+    The StatsD monitor can parse keywords from a statsd metric name by a set of
+    converter that was configured by user.
+
+    ```
+    converters:
+      - pattern: "cluster.cds_{traffic}_{mesh}_{service}-vn_{protocol}_{port}.{action}"
+        dimensions:
+          - traffic
+          - mesh
+          - service
+          - action
+    ```
+
+    This converter will parse `traffic`, `mesh`, `service` and `action` as dimensions
+    from a metric name `cluster.cds_egress_ecommerce-demo-mesh_gateway-vn_tcp_8080.update_success`.
+
+    Metrics will be sent without dimensions if there's no related configuration.
+
+    #### Formatting metric name
+
+    You can customize a metric name by providing a format string within the converter configuration.
+    
+    ```
+    converters:
+      - pattern: "cluster.cds_{traffic}_{mesh}_{service}-vn_{protocol}_{port}.{action}"
+        metric: "{traffic}.{action}"
+    ```
+    
+    The metrics which match to the given pattern will be reported to SignalFx as `{traffic}.{action}`.
+    For instance, metric `cluster.cds_egress_ecommerce-demo-mesh_gateway-vn_tcp_8080.update_success`
+    will be reported as `egress.update_success`.
+
+    The original metric name will be sent without being processed if there's no related configuration.
```

Comment:
Don't allow the original metric name to be sent if there is a pattern specified.  Force the user to specify how to create a metric name in that case.

If no configuration at all (no `pattern` or `metric`), only then should the metric be passed through as is.

Reply:
That would essentially mean make a metric name for a converter required. I'll do so.

Reply:
Docs don't seem to indicate that metricName is required.